app/src/main/java/org/groundplatform/android/MainViewModel.kt

```diff
@@ -126,6 +142,18 @@ constructor(
       userRepository.saveUserDetails(user)
       if (!isTosAccepted()) {
         MainUiState.TosNotAccepted
+      } else if (isDeepLinkAvailable()) {
+        val deepLinkUri = _deepLinkUri.value
+        val surveyId =
+          deepLinkUri
+            ?.takeIf { it.pathSegments.firstOrNull() == SURVEY_PATH_SEGMENT }
+            ?.lastPathSegment
+
+        if (!surveyId.isNullOrBlank()) {
+          MainUiState.ActiveSurveyById(surveyId)
+        } else {
+          MainUiState.NoActiveSurvey
```

Comment:
Should we also log an error with the uri?

Reply:
not an error